### PR TITLE
Enable encoder settings VIA.

### PR DESCRIPTION
**Pull request by 7-rate**
## Description

The same action as #14599 will be taken for pistachio_pro.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pistachio_pro/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/pistachio_pro/keymaps/via/keymap.c:168]:
```suggestion
}

void encoder_action_register(uint8_t index, bool clockwise) {
    keyevent_t encoder_event = (keyevent_t) {
        .key = clockwise ? ENC_CW : ENC_CCW,
        .pressed = true,
        .time = (timer_read() | 1)
    };
    encoder_state = (clockwise ^ 1) | (clockwise << 1);
    action_exec(encoder_event);
}

void matrix_scan_user(void) {
    encoder_action_unregister();
}

bool encoder_update_user(uint8_t index, bool clockwise) {
    encoder_action_register(index, clockwise);
    return false;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by 7-rate** [comment]
If there are no problems, I would like you to merge them.

**Comment by drashna** [comment]
> If there are no problems, I would like you to merge them.

Just a heads up, we're in a merge freeze, ATM, due to the breaking changes cycle. 
https://docs.qmk.fm/#/breaking_changes?id=_2-days-before-merge

Also, in general, we require 2 approvals before a PR is merged.

**Review by tzarc — APPROVED** [review]

**Pull request merged**
